src/server/system_services/system_server.js

```diff
@@ -1282,6 +1282,24 @@ async function _get_endpoint_groups() {
     });
 }
 
+/**
+ *
+ * ROTATE ROOT KEY
+ *
+ */
+async function rotate_root_key(req) {
+    const new_secret = req.rpc_params.new_secret;
+    const mkm = system_store.master_key_manager;
+    await P.all(_.map(system_store.data.systems, async function(system) {
+        const reencrypted = mkm._reencrypt_master_key_by_root(system.master_key_id._id, new_secret);
+        await upsert_master_key({
+            _id: system.master_key_id._id,
+            update: { cipher_key: reencrypted }
+        });
+        _rotate_master_key(system.name, 'SYSTEM');
```

Comment:
Don't we want to keep the old root key in some way?

Reply:
not as part of core - not sure if will be saved by operator also